### ISSUE TITLE
When initializing Ray with ray init, the pip source can be set in the runtime_env.

### ISSUE THREAD

**Issue by Jay-ju** [ray-project/ray#49247]
### Use case

ray.init(runtime_env={"pip": ["wget", "zipfile37",
         "vllm"]}, ignore_reinit_error=True)

**Comment by jjyao** [comment]
@Jay-ju do you want to specify a pip source other than pypi? You can already do that today: You can define your requirements in a `requirement.txt` file and specify `--index-url` or `--extra-index-url` options. See https://docs.ray.io/en/latest/ray-core/handling-dependencies.html#using-conda-or-pip-packages

**Comment by Jay-ju** [comment]
> @Jay-ju do you want to specify a pip source other than pypi? You can already do that today: You can define your requirements in a `requirement.txt` file and specify `--index-url` or `--extra-index-url` options. See https://docs.ray.io/en/latest/ray-core/handling-dependencies.html#using-conda-or-pip-packages
Yes, I really need to set the pip source mainly because of some network issues. If I were to set the requirements.txt, I'd have to change the mirror environment. What I hope is to install packages directly without changing the mirror.
 Here, I have some ideas. We can define a format for the libraries to be installed. For example, vllm==2.2@https://pypi.tuna.tsinghua.edu.cn/simple. I can parse the content after the @ symbol and assemble it into a pip command. Is this idea feasible? https://pypi.tuna.tsinghua.edu.cn/simple
